routes/claims/claims.controller.js

```diff
@@ -0,0 +1,50 @@
+const { validationResult, checkSchema } = require('express-validator')
+const { errorArray2ErrorObject } = require('./../../utils.js')
+//const { loginSchema, sinSchema } = require('./../../formSchemas.js')
+const API = require('../../api')
+
+module.exports = function (app) {
+    // redirect from "/deductions" → "/login/code", you really shouldn't be here (Unless we add a "info" page)
+    app.get('/deductions', (req, res) => res.redirect('/login/code'))
+
+    app.get('/deductions/rrsp', (req, res) => res.render('deductions/rrsp', { data: req.session || {} }))
+    app.post('/deductions/rrsp', checkSchema(loginSchema), postRRSP)
+
+}
+
+//POST functions that handle setting the login data in the session and handle redirecting to the next page or sending an error to the client.
+//Note that this is not the only error validation, see routes defined above.
+const validateRedirect = req => {
+    let redirect = req.body.redirect || null
+    if (!redirect) {
+        throw new Error(`[POST ${req.path}] 'redirect' parameter missing`)
+    }
+    return redirect
+}
+
+const postRRSP = (req, res) => {
+    const redirect = validateRedirect(req, res)
+
+    const errors = validationResult(req)
+
+    //If sin is not set, set it to null
+    let rrsp = req.body.rrsp || null
+    req.session.credits.rrspClaim = rrsp
```

Comment:
Is this going to be `deductions` now, instead of credits?

Reply:
yes